src/block-data-providers/esplora.provider.ts

```diff
@@ -162,33 +162,59 @@ export class EsploraProvider
     }
 
     private async processBlock(height: number, hash: string) {
+        const state = await this.getState();
+        const lastProcessedTxIndex = state?.lastProcessedTxIndex ?? 0;
```

Comment:
Assume that it's defined
```suggestion
        const lastProcessedTxIndex = state.lastProcessedTxIndex;
```